cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -92,7 +97,13 @@ public BasicMapReduceContext(Program program,
     } else {
       this.userMetrics = null;
     }
-    this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    String adapterName = getAdapterName(runtimeArguments);
+    if (adapterName == null) {
+      this.loggingContext = new MapReduceLoggingContext(getNamespaceId(), getApplicationId(), getProgramName());
+    } else {
+      this.loggingContext = new BatchAdapterLoggingContext(getNamespaceId(), getApplicationId(), getProgramName(),
```

Comment:
I was initially under the impression that adapter logs needed to be in their own partition/own directory etc. However, doesn't look like that is needed. Removed ``BatchAdapterLoggingContext``. Added an optional ``.adapterId`` tag in ``ApplicationLoggingContext``, that various programs can set if required.